src/qscand.cpp

```diff
@@ -0,0 +1,372 @@
+/*
+ * This file is part of the QPxTool project.
+ * Copyright (C) 2008-2009 Gennady "ShultZ" Kozlov <qpxtool@mail.ru>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ * See the file "COPYING" for the exact licensing terms.
+ *
+ */
+
+#include <cerrno>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <unistd.h>
+#include <getopt.h>
+
+#include "common_functions.hpp"
+#include "threads.hpp"
+#include "child.hpp"
+#include "qscand.hpp"
+#ifndef _WIN32
+#include <signal.h>
+#endif
+
+#include <fcntl.h>
+
+bool term;
+int childl;
+pid_t pid;
+// pid_t cpids[CLIENTS_MAX];
+bool daemonized=0;
+bool debug=0;
+
+#if defined (__unix) || defined (__unix__)
+void sigint_handler (int)
+#elif defined (_WIN32)
+BOOL WINAPI sigint_handler (DWORD)
+#endif
+{
+	if (!childl) {
+		if (debug && !daemonized && clients>0)
+			printf("%d: SIGINT. %d client(s) left. Terminating connections...\n", pid, clients);
+//		while (clients) msleep((1 << 8));
+//		if (debug && !daemonized)
+//			printf("%d: SIGINT. Terminating listener...\n", pid);
+	} else if (childl == 1) {
+		if (debug && !daemonized)
+			printf("%d: SIGINT. Terminating client connection...\n", pid);
+	}
+	term=1;
+#if defined (_WIN32)
+	return 1;
+#endif
+}
+
+#ifdef DAEMON_EN
+int daemonize()
+{
+	pid_t pid;
+	switch (pid = fork()) {
+		case -1:
+			return -1;
+		case 0:
+			openlog("qscand", LOG_PID, LOG_DAEMON);
+			close(0);
+			close(1);
+			close(2);
+			daemonized = 1;
+			break;
+		default:
+			exit(0);
+	}
+	if (setsid() == -1)
+		return -1;
+	return 0;
+}
+#endif
+
+static struct option long_options[] = {
+	{"help",      0, NULL, 'h'},
+#ifdef DAEMON_EN
+	{"nodaemon",  0, NULL, 'n'},
+#endif
+	{"debug",     0, NULL, 'd'},
+	{"iface",     1, NULL, 'i'},
+	{"port",      1, NULL, 'p'},
+#ifdef AUTOPORT_ENABLE
+	{"portauto",  0, NULL, 'a'},
+#endif
+	{0,0,0,0}
+};
+	
+int child_create(int idx)
+{
+	int r;
+	if ((r = thread_create(&childs[idx].tid, NULL, &child_thread, (void*)&childs[idx].arg))) {
+#ifdef DAEMON_EN
+		syslog(LOG_ERR, "Error creating thread!\n");
+#else
+		printf("Error creating thread!\n");
+#endif
+		return r;
+		close(childs[idx].arg.connfd);
```

Comment:
Codacy found an issue: [Statements following return, break, continue, goto or throw will never be executed.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6191247)